src/Razor/src/Microsoft.NET.Sdk.Razor/GenerateStaticWebAssetsManifest.cs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class GenerateStaticWebAssetsManifest : Task
+    {
+        private const string ContentRoot = "ContentRoot";
+        private const string BasePath = "BasePath";
+
+        [Required]
+        public string TargetManifestPath { get; set; }
+
+        [Required]
+        public ITaskItem[] ContentRootDefinitions { get; set; }
+
+        public override bool Execute()
+        {
+            if (!ValidateArguments())
+            {
+                return false;
+            }
+
+            return ExecuteCore();
+        }
+
+        private bool ExecuteCore()
+        {
+            var document = new XDocument(new XDeclaration("1.0", "utf-8", "yes"));
+            var root = new XElement(
+                "StaticWebAssets",
+                new XAttribute("Version", "1.0"),
+                CreateNodes());
+
+            document.Add(root);
+
+            var settings = new XmlWriterSettings
+            {
+                Encoding = Encoding.UTF8,
+                CloseOutput = true,
+                OmitXmlDeclaration = true,
+                Indent = true,
+                NewLineOnAttributes = false,
+                Async = true
+            };
+
+            using (var xmlWriter = GetXmlWriter(settings))
+            {
+                document.WriteTo(xmlWriter);
+            }
+
+            return true;
```

Comment:
`return !Log.HasLoggedErrors;`

Comes highly recommended

Reply:
What's the advantage? If we got here I would expect unconditional success.

Reply:
Done